From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Don't use connection assumption.
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/network-manager/+bug/963106
Forwarded: not-needed

Basically, we want dhclient to be killed for now whenever NetworkManager gets
stopped, because otherwise filesystems may be holding lease files around
when they get unmounted. NetworkManager gets killed very late in the shutdown
process, even after sendsigs gets run (Sending TERM to all processes...)
which means we can't rely on that to kill any dhclient processes. NM must
do the reaping on its own.

A side effect of this may be slightly slower bringinging up time for devices;
when they are in a configuration that would match something NM knows about
in configuration. However, this only applied to IPv4 since IPv6 connection
assumption isn't implemented either.

Reason for this is that we value more correct behavior and proper shutdown
without unmounting filesystems in a dirty state, than trying to be clever
and short-circuiting device configuration for a few ms faster start when
NM gets restarted.

---
 src/nm-device.c  |    2 ++
 src/nm-manager.c |    2 ++
 2 files changed, 4 insertions(+)

Index: b/src/nm-device.c
===================================================================
--- a/src/nm-device.c
+++ b/src/nm-device.c
@@ -3801,6 +3801,7 @@ dispose (GObject *object)
 
 	priv->disposed = TRUE;
 
+#ifndef TARGET_DEBIAN
 	/* Don't down can-assume-connection capable devices that are activated with
 	 * a connection that can be assumed.
 	 */
@@ -3825,6 +3826,7 @@ dispose (GObject *object)
 				take_down = FALSE;
 		}
 	}
+#endif /* TARGET_DEBIAN */
 
 	/* Clear any queued transitions */
 	nm_device_queued_state_clear (self);
Index: b/src/nm-manager.c
===================================================================
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -525,6 +525,7 @@ remove_one_device (NMManager *manager,
 	NMManagerPrivate *priv = NM_MANAGER_GET_PRIVATE (manager);
 
 	if (nm_device_get_managed (device)) {
+#ifndef TARGET_DEBIAN
 		/* When quitting, we want to leave up interfaces & connections
 		 * that can be taken over again (ie, "assumed") when NM restarts
 		 * so that '/etc/init.d/NetworkManager restart' will not distrupt
@@ -536,6 +537,7 @@ remove_one_device (NMManager *manager,
 		if (   !nm_device_can_assume_connections (device)
 		    || (nm_device_get_state (device) != NM_DEVICE_STATE_ACTIVATED)
 		    || !quitting)
+#endif
 			nm_device_set_managed (device, FALSE, NM_DEVICE_STATE_REASON_REMOVED);
 	}
 
